src/annotator/guest.coffee

```diff
@@ -451,51 +511,6 @@ module.exports = class Guest extends Delegator
   isEventInAnnotator: (event) ->
     return closest(event.target, '.annotator-frame') != null
 
-  # Event handlers to close the sidebar when the user clicks in the document.
-  # These really ought to live with the sidebar code.
-  onElementClick: (event) ->
-    if this.closeSidebarOnDocumentClick
-      if !this.isEventInAnnotator(event) and !@selectedTargets?.length and not event.annotations?.length
-        @crossframe?.call('hideSidebar')
-
-  onElementTouchStart: (event) ->
-    # Mobile browsers do not register click events on
-    # elements without cursor: pointer. So instead of
-    # adding that to every element, we can add the initial
-    # touchstart event which is always registered to
-    # make up for the lack of click support for all elements.
-    if this.closeSidebarOnDocumentClick
-      if !this.isEventInAnnotator(event) and !@selectedTargets?.length and not event.annotations?.length
-        @crossframe?.call('hideSidebar')
-
-  onHighlightMouseover: (event) ->
-    return unless @visibleHighlights
-    annotation = $(event.currentTarget).data('annotation')
-    annotations = event.annotations ?= []
-    annotations.push(annotation)
-
-    # The innermost highlight will execute this.
-    # The timeout gives time for the event to bubble, letting any overlapping
-    # highlights have time to add their annotations to the list stored on the
-    # event object.
-    if event.target is event.currentTarget
-      setTimeout => this.focusAnnotations(annotations)
-
-  onHighlightMouseout: (event) ->
-    return unless @visibleHighlights
-    this.focusAnnotations []
-
-  onHighlightClick: (event) ->
-    return unless @visibleHighlights
-    annotation = $(event.currentTarget).data('annotation')
-    annotations = event.annotations ?= []
-    annotations.push(annotation)
-
-    # See the comment in onHighlightMouseover
-    if event.target is event.currentTarget
-      xor = (event.metaKey or event.ctrlKey)
-      setTimeout => this.selectAnnotations(annotations, xor)
```

Comment:
Previously the code would "annotate" click events on highlights with the associated annotation as the event bubbled up through the document. The timeout was needed to trigger selecting _all_ of the annotations associated with highlights that the event bubbled through. In the new code, this is handled in a more direct way by iterating up the DOM tree from the element that was originally clicked and creating a list of all `<hypothesis-highlight>` elements found along the way.